src/OpenLoco/src/Vehicles/VehicleHead.cpp

```diff
@@ -1925,7 +1917,7 @@ namespace OpenLoco::Vehicles
             status = Status::travelling;
             status = sub_427BF2();
             advanceToNextRoutableOrder();
-            if (!(updateWaterMotion(0) & WaterMotionFlags::hasReachedDock))
+            if (!hasFlags(updateWaterMotion(WaterMotionFlags::none), WaterMotionFlags::hasReachedDock))
```

Comment:
```suggestion
            if (auto newFlags = updateWaterMotion(WaterMotionFlags::none); 
                (newFlags & WaterMotionFlags::hasReachedDock) != WaterMotionFlags::None)
```
I think if you do it like that you can avoid the need of hasFlags